hunts/views.py

```diff
@@ -93,17 +93,21 @@ def post(self, request, pk):
             # TODO(asdfryan): Add error handling and refactor into slack lib.
             slack_client = SlackClient.getInstance()
             channel_id = slack_client.create_or_join_channel(name)
+            if channel_id is None:
+                messages.warning(request, "Slack channel not created")
 
             try:
                 puzzle_class.objects.create(
                     name=name,
                     url=puzzle_url,
                     hunt=hunt,
                     sheet=sheet,
-                    channel=channel_id
+                    channel=channel_id if channel_id else name
                 )
                 # Announce new puzzle is available on slack.
-                slack_client.announce_puzzle_creation(name, channel_id, is_meta)
+                if channel_id:
+                    slack_client.announce_puzzle_creation(name, channel_id, is_meta)
```

Comment:
Hmm ideally slack_client.announce_puzzle_creation() should just do nothing when slack API token is missing. I think I'd rather make that fix instead.